tests/integration/test_charm.py

```diff
@@ -118,3 +128,99 @@ async def test_add_admin(app: Application):
     assert action.status == "completed"
     assert action.results["user"] == email
     assert f'Admin with email "{email}" correctly created' in action.results["output"]
+
+
+@pytest.mark.asyncio
+@pytest.mark.abort_on_fail
+@pytest.mark.requires_secrets
+async def test_saml_auth(
+    ops_test: OpsTest,
+    app: Application,
+    saml_email: str,
+    saml_password: str,
+    requests_timeout: float,
+):
+    """
+    arrange: given charm in its initial state
+    act: configure a SAML target url and fire SAML authentication
+    assert: The SAML authentication process is executed successfully.
+    """
+    await app.set_config(  # type: ignore[attr-defined] # pylint: disable=W0106
+        {
+            "site_url": "https://indico.local",
+            "saml_target_url": STAGING_UBUNTU_SAML_URL,
+        }
+    )
+    await ops_test.model.wait_for_idle(status="active")  # type: ignore[union-attr]
+    urllib3.disable_warnings(urllib3.exceptions.InsecureRequestWarning)
+
+    host = "indico.local"
+    original_getaddrinfo = socket.getaddrinfo
+
+    def patched_getaddrinfo(*args):
+        if args[0] == host:
+            return original_getaddrinfo("127.0.0.1", *args[1:])
+        return original_getaddrinfo(*args)
+
+    with patch.multiple(socket, getaddrinfo=patched_getaddrinfo):
```

Comment:
```suggestion
with patch.multiple(socket, getaddrinfo=patched_getaddrinfo), requests.session() as session:
```
Would this syntax be more readable?